pyproject.toml

```diff
@@ -74,4 +76,4 @@ testpaths = [
 
 [tool.coverage.run]
 branch = true
-source = ["stmtools"]
\ No newline at end of file
+source = ["stmtools"]
```

Comment:
Default ruff configuration assumes `python 3.8,` see t`arget-version` [here](https://github.com/astral-sh/ruff/blob/main/docs/configuration.md).  I suggest adding the same configs [here](https://github.com/EXCITED-CO2/xarray-regrid/blob/d4980b9108d68ed70ca5b337ac8ca9229252b504/pyproject.toml#L80C1-L80C12) and [here](https://github.com/EXCITED-CO2/xarray-regrid/blob/d4980b9108d68ed70ca5b337ac8ca9229252b504/pyproject.toml#L131) to the `pyproject.toml`.

Reply:
I applied the python 3.10 rule and kept others the same

Reply:
As explained [here](https://docs.astral.sh/ruff/configuration/#config-file-discovery), if you want to use` pyproject.toml`, you need to pass all the settings because ruff will use this file instead of the default file, ruff does not merge settings across different files. So I suggest adding the same configs mentioned [here](https://github.com/EXCITED-CO2/xarray-regrid/blob/d4980b9108d68ed70ca5b337ac8ca9229252b504/pyproject.toml#L80C1-L80C12) and [here](https://github.com/EXCITED-CO2/xarray-regrid/blob/d4980b9108d68ed70ca5b337ac8ca9229252b504/pyproject.toml#L131) to the `pyproject.toml`.